### PR TITLE
LPS-189023 | Poshi Automation for Bug | Master

### PR DESCRIPTION
**Pull request by rafaellajordao**
# LPS-189023 | Poshi Automation for Bug

Hi! :)
This pr is corresponding to this ticket: [LPS-188364](https://liferay.atlassian.net/browse/LPS-188364) → [LPS-189023](https://liferay.atlassian.net/browse/LPS-189023)

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by rafaellajordao** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsConfiguration.testcase [hunk 1]
```diff
@@ -190,6 +190,54 @@ element.textContent = 'Fragment Example';''';
...
+			AssertVisible(
+				key_fragmentName = "Menu Display",
+				locator1 = "Fragment#FRAGMENT_HEADER_ELLIPSIS");
+
+			Click.waitForMenuToggleJSClick(
+				key_fragmentName = "Menu Display",
+				locator1 = "Fragment#FRAGMENT_HEADER_ELLIPSIS");
+
+			MenuItem.click(menuItem = "Delete");
+		}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsConfiguration.testcase:238]: Should be Menu Display widget on header

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsConfiguration.testcase [hunk 2]
```diff
@@ -190,6 +190,54 @@ element.textContent = 'Fragment Example';''';
 		}
 	}
 
+	@description = "This is a test for LPS-188364. Unable to configure header search bar or menu display on content pages."
+	@priority = 4
+	test CanConfigureSerachAndMenuDisplayWidgets {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsConfiguration.testcase:195]: ConfigureMenuDisplayWidget

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsConfiguration.testcase [hunk 3]
```diff
@@ -190,6 +190,54 @@ element.textContent = 'Fragment Example';''';
 		}
 	}
 
+	@description = "This is a test for LPS-188364. Unable to configure header search bar or menu display on content pages."
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsConfiguration.testcase:193]: The user is able to configure Menu Display on content page. Migrate to the NavigationMenuWidgets.testcase

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsConfiguration.testcase [hunk 4]
```diff
@@ -190,6 +190,54 @@ element.textContent = 'Fragment Example';''';
...
+	@priority = 4
+	test CanConfigureSerachAndMenuDisplayWidgets {
+		task ("Given a content page") {
+			ContentPagesNavigator.openEditContentPage(
+				pageName = "Test Page Name",
+				siteName = "Test Site Name");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsConfiguration.testcase:199]: Add a site based on Welcome then navigate to the Home page

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsConfiguration.testcase [hunk 5]
```diff
@@ -190,6 +190,54 @@ element.textContent = 'Fragment Example';''';
...
+			AssertVisible(
+				key_fragmentName = "Search",
+				locator1 = "Fragment#FRAGMENT_HEADER_ELLIPSIS");
+
+			Click.waitForMenuToggleJSClick(
+				key_fragmentName = "Search",
+				locator1 = "Fragment#FRAGMENT_HEADER_ELLIPSIS");
+
+			MenuItem.click(menuItem = "Delete");
+		}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsConfiguration.testcase:218]: Should be search bar widget on header but Menu Display is already enough

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsConfiguration.testcase [hunk 6]
```diff
@@ -190,6 +190,54 @@ element.textContent = 'Fragment Example';''';
...
+				key_fragmentName = "Menu Display",
+				locator1 = "Fragment#FRAGMENT_HEADER_ELLIPSIS");
+
+			Click.waitForMenuToggleJSClick(
+				key_fragmentName = "Menu Display",
+				locator1 = "Fragment#FRAGMENT_HEADER_ELLIPSIS");
+
+			MenuItem.click(menuItem = "Delete");
+		}
+	}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsConfiguration.testcase:239]: Then add steps like
NavigationMenusWidget.gotoNavigationMenuWidgetConfig();
Make some changes

**Review by Tim-Cao — CHANGES_REQUESTED** [review]
Hi @rafaellajordao, would you mind modifing according to comments? Thanks

**Comment by rafaellajordao** [comment]
Hi @Tim-Cao !
I couldn't add the Menu Display directly to the predefined Home Page, if it's incorrect can you show me how I can do it? :)

**Comment by Tim-Cao** [comment]
Hi @rafaellajordao, Yes, we don't need to add it. Just use the OOTB one on page.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/sitenavigation/navigationmenu/NavigationMenuWidget.testcase [hunk 7]
```diff
@@ -248,6 +248,50 @@ definition {
...
+	@priority = 4
+	test ConfigureMenuDisplayWidget {
+		task ("Given a content page") {
+			JSONLayoutpagetemplate.addMasterPageTemplateEntry(
+				groupName = "Test Site Name",
+				masterPageTemplateEntryName = "Masters Page Template");
+
+			MastersPageTemplatesAdmin.openMastersAdmin(siteURLKey = "test-site-name");
+
+			MastersPageTemplatesAdmin.gotoMasters(masterLayoutName = "Masters Page Template");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/sitenavigation/navigationmenu/NavigationMenuWidget.testcase:261]: Add a site based on Welcome

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/sitenavigation/navigationmenu/NavigationMenuWidget.testcase [hunk 8]
```diff
@@ -248,6 +248,50 @@ definition {
...
+				collectionName = "Navigation Bars",
+				dropTarget = "Header",
+				fragmentName = "Header Light",
+				scope = "Master");
+
+			PageEditor.addWidget(
+				collectionName = "Content Management",
+				dropTarget = "Header",
+				portletName = "Menu Display",
+				scope = "Master");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/sitenavigation/navigationmenu/NavigationMenuWidget.testcase:275]: When the user accesses to the Home page

**Review by Tim-Cao — CHANGES_REQUESTED** [review]

**Comment by Tim-Cao** [comment]
See changes @rafaellajordao
https://github.com/liferay-echo/liferay-portal/pull/13203

**Comment by Tim-Cao** [comment]
cI:close

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**